static/index.js

```diff
@@ -0,0 +1,294 @@
+      case "hunger":
+        setHunger(message);
+        break;
+      case "started":
+        botOnline(message.action);
+        break;
+    }
+  };
+  socketserver.onclose = function (event) {
+    console.log("connection closed");
+    serverOnline(false);
+  };
+}
+
+// functions
+
+function say(message) {
+  send({
+    action: "say",
+    botId: getCurrentBot(),
+    message: message,
+  });
+}
+function kill() {
+  send({
+    action: "kill",
+    botId: getCurrentBot(),
+  });
+}
+
+function place() {
+  send({
+    action: "place",
+    botId: getCurrentBot(),
+  });
+}
+function attack() {
+  send({
+    action: "attack",
+    botId: getCurrentBot(),
+  });
+}
+function setServer(address, port) {
+  var address = document.getElementById("server").value;
+  var port = document.getElementById("port").value;
+  var data = { address: address, port: port };
+
+  send({
+    action: "setServer",
+    botId: getCurrentBot(),
+    data: data,
+  });
+}
+function stopBot() {
+  send({
+    action: "stop",
+    botId: getCurrentBot(),
+  });
+}
+function startBot() {
+  send({
+    action: "start",
+    botId: getCurrentBot(),
+  });
+}
+function getCoords() {
+  if (connected) {
+    send({
+      action: "getCoords",
+      botId: getCurrentBot(),
+    });
+  }
+}
+function getHealth() {
+  if (connected) {
+    send({
+      action: "getHealth",
+      botId: getCurrentBot(),
+    });
+  }
+}
+function getHunger() {
+  if (connected) {
+    send({
+      action: "getHunger",
+      botId: getCurrentBot(),
+    });
+  }
+}
+
+function setHunger(message) {
+  document.getElementById("hunger").innerHTML = message.data;
+}
+
+function setCoords(message) {
+  document.getElementById("coords").innerHTML =
+    Math.round(message.data.x) +
+    " " +
+    Math.round(message.data.y) +
+    " " +
+    Math.round(message.data.z);
+}
+function setHealth(message) {
+  document.getElementById("health").innerHTML = message.data;
+}
+
+function getCurrentBot() {
+  return document.getElementById("botSelected").value;
+}
+
+/*
+required headers
+action:
+botId:
+
+*/
+function send(message) {
+  JSON.stringify(message);
+  socketserver.send(JSON.stringify(message));
+}
+
+function closeWebSocket() {
+  socketserver.close();
+}
+function startWebSocket() {
+  socketserver = new WebSocket("ws://0.0.0.0:3000", "protocolOne");
+  listenSocket();
+}
+
+function toggleTheme() {
+  document.body.classList.toggle("dark-body");
+  for (
+    let index = 0;
+    index < document.getElementsByClassName("card").length;
+    index++
+  ) {
+    document
+      .getElementsByClassName("card")
+      .item(index)
+      .classList.toggle("bg-dark");
+  }
+}
+
+function serverOnline(state) {
+  var status = document.getElementById("serverStatus");
+  if (state) {
+    status.setAttribute("class", "badge badge-success");
+    status.innerHTML = "Online";
+  } else {
+    status.setAttribute("class", "badge badge-danger");
+    status.innerHTML = "Offline";
+  }
+}
```

Comment:
There is no element with id "serverStatus"

Reply:
cuz i got rid of it